ext/crypto/lib.rs

```diff
@@ -458,8 +465,13 @@ pub async fn op_crypto_verify_key(
 
   let verification = match algorithm {
     Algorithm::RsassaPkcs1v15 => {
-      let public_key: RsaPublicKey =
-        RsaPrivateKey::from_pkcs1_der(&*args.key.data)?.to_public_key();
+      let public_key = match args.key.r#type {
+        KeyType::Private => {
+          RsaPrivateKey::from_pkcs1_der(&*args.key.data)?.to_public_key()
+        }
+        KeyType::Public => RsaPublicKey::from_pkcs1_der(&*args.key.data)?,
+        _ => unreachable!(),
```

Comment:
```suggestion
        KeyType::Secret => unreachable!("unexpected KeyType::Secret"),
```
That way, if another key type is added later, the compiler will complain about non-exhaustive checks, making it easy to audit each use site.

Applies here and elsewhere.